examples/all-clusters-app/all-clusters-common/src/mode-select-delegates.cpp

```diff
@@ -0,0 +1,171 @@
+
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        response.status = int8_t(ChangeToModeResponseStatus::kGenericFailure);
+        response.statusText.SetValue(chip::CharSpan("Could not get the current mode", 30));
+        return;
+    }
+
+    // Our business logic states that we can only switch into the mapping state from the idle state.
+    if (mode == 2 && currentMode != 0)
+    {
+        response.status = int8_t(ChangeToModeResponseStatus::kGenericFailure);
+        response.statusText.SetValue(chip::CharSpan("Change to the mapping state is only allowed from idle", 53));
+    }
+
+    response.status = int8_t(ChangeToModeResponseStatus::kSuccess);
+}
+
+//-- RVC Clean delegate functions
+CHIP_ERROR RvcCleanDelegate::Init()
+{
+    return CHIP_NO_ERROR;
+}
+
+Status RvcCleanDelegate::HandleChangeToMode(uint8_t mode)
+{
+    return Status::UnsupportedCommand;
+}
+
+void RvcCleanDelegate::HandleChangeToModeWitheStatus(uint8_t mode, Commands::ChangeToModeResponse::Type &response)
+{
+    uint8_t currentMode;
+    EmberAfStatus status = RvcRun::Attributes::CurrentMode::Get(0x1, &currentMode);
+
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        response.status = int8_t(ChangeToModeResponseStatus::kGenericFailure);
+        response.statusText.SetValue(chip::CharSpan("Could not get the current mode", 30));
+        return;
+    }
+
+    if (currentMode == 1)
+    {
+        response.status = int8_t(RvcClean::ChangeToModeResponseStatus::kCleaningInProgress);
+        response.statusText.SetValue(chip::CharSpan("Cannot change the cleaning mode during a clean", 60));
+    }
+
+    response.status = int8_t(ChangeToModeResponseStatus::kSuccess);
+}
+
+//-- Dishwasher Control delegate functions
+CHIP_ERROR DishwasherControlDelegate::Init()
+{
+    return CHIP_NO_ERROR;
+}
+
+Status DishwasherControlDelegate::HandleChangeToMode(uint8_t mode)
+{
+    return Status::Success;
+}
+
+void DishwasherControlDelegate::HandleChangeToModeWitheStatus(uint8_t mode, Commands::ChangeToModeResponse::Type &response)
+{
+    response.status = int8_t(ChangeToModeResponseStatus::kSuccess);
+}
+
+//-- Laundry Washer delegate functions
+CHIP_ERROR LaundryWasherDelegate::Init()
+{
+    return CHIP_NO_ERROR;
+}
+
+Status LaundryWasherDelegate::HandleChangeToMode(uint8_t mode)
+{
+    return Status::Success;
+}
+
+void LaundryWasherDelegate::HandleChangeToModeWitheStatus(uint8_t mode, Commands::ChangeToModeResponse::Type &response)
+{
+    response.status = int8_t(ChangeToModeResponseStatus::kSuccess);
+}
+
+//-- Refrigerator And Temperature Controlled Cabinet delegate functions
+CHIP_ERROR TccDelegate::Init()
+{
+    return CHIP_NO_ERROR;
+}
+
+Status TccDelegate::HandleChangeToMode(uint8_t mode)
+{
+    return Status::Success;
+}
+
+void TccDelegate::HandleChangeToModeWitheStatus(uint8_t mode, Commands::ChangeToModeResponse::Type &response)
+{
+    response.status = int8_t(ChangeToModeResponseStatus::kSuccess);
```

Comment:
All statuses should be uint8_t, not int8_t. Not sure why this is int8_t. Signedness is not used.

Reply:
Good spot! It was a mistake on my part. Fixed in commit [d616e4b7823c2dab11845c55dfcbf2ae4ae3c773](https://github.com/project-chip/connectedhomeip/pull/26447/commits/d616e4b7823c2dab11845c55dfcbf2ae4ae3c773).